code/modules/urist/modules/GOAI/integrations/base_commander/mob_commander/actions/waypoint.dm

```diff
@@ -91,13 +91,13 @@
 			var/last_link_blocked = GoaiLinkBlocked(previous, pathitem)
 
 			if(last_link_blocked)
-				OBSTACLEHUNT_DEBUG_LOG("[owner]: LINK BETWEEN [previous] & [pathitem] OBSTRUCTED")
+				OBSTACLEHUNT_DEBUG_LOG("[owner]: LINK BETWEEN [previous] ([previous.x],[previous.y]) & [pathitem] ([pathitem.x],[pathitem.y]) OBSTRUCTED")
 				// find the obstacle
 				var/atom/obstruction = null
 
 				if(!obstruction)
 					for(var/atom/potential_obstruction_curr in pathitem.contents)
-						var/datum/directional_blocker/blocker = potential_obstruction_curr?.directional_blocker
+						var/datum/directional_blocker/blocker = potential_obstruction_curr.GetBlockerData(TRUE, TRUE)
```

Comment:
As a rule, I'd recommend the null-conditional `?.` here to deal with nulls.

Reply:
Is`?.` necessary here? Doesn't the way DM iterate through lists like this ignore nulls because we're iterating by type `/atom`? If it's not of type `/atom`, it's not iterated over surely?

Reply:
Hmm... that's an interesting question and I don't know the answer to it. As a rule of thumb, I don't trust anything DM gives me as non-null until explicitly checked, but it might be unnecessary.

Reply:
From DM reference:
```If the loop var has a type, a hidden istype() call is included in the code. Only items of that type, or its descendants, are handled within the loop. That means null values and objects of unrelated types will be skipped.```

Reply:
Of-bloody-course it's an _implicit typecheck_, can't have any too-clever optimizations in this godforsaken language...

![image](https://user-images.githubusercontent.com/4363639/212190745-072d6d71-2c2c-4451-a16b-56adcb35ba54.png)